app/src/main/java/com/simplecity/amp_library/playback/MusicService.java

```diff
@@ -1047,7 +1052,27 @@ void saveQueue(final boolean full) {
                         }
                     }
                 }
-                editor.putString("shuffleList", q.toString());
+                editor.putString("trackShuffleList", q.toString());
+
+            } else if (shuffleMode == ShuffleMode.ALBUMS) {
+                len = albumShuffleList.size();
+                q.setLength(0);
+                for (int i = 0; i < len; i++) {
```

Comment:
This could be factored out into a method, since building the list of reverse-hex numbers is now done in the same fashion for three separate lists.